dts refactor

ktaborowski opened the pull request:
Use dts and shield.
Compatibility with Zephyr Lora Basic Modem project.

To do:
- [x] nRF54L support
- [x] nRF53, nRF52 support
- [ ] ...

## CI parameters

```yaml
Jenkins:
  test-sdk-sidewalk: master
  # To reconfigure functional tests:
  # use GH labels func-* (default is func-commit)
  # or
  # Use YAML Filters. Helper side to set filters: https://tester-pc.nordicsemi.no:8080/test_mgmt
  # Filters (place copied YAML filters here):
  #  - filter1:
  #     board: nrf52
```

## Description

JIRA ticket: 

## Self review

- [ ] There is no commented code.
- [ ] There are no TODO/FIXME comments without associated issue ticket.
- [ ] Commits are properly organized.
- [ ] Change has been tested.
- [ ] Tests were updated (if applicable).

ktaborowski commented on boards/shields/nrf54l15dk_arduino_adapter/nrf54l15dk_arduino_adapter.overlay:
```diff
@@ -0,0 +1,141 @@
+        #gpio-cells = <2>;
+        gpio-map-mask = <0xffffffff 0xffffffc0>;
+        gpio-map-pass-thru = <0 0x3f>;
+        gpio-map = <0 0 &gpio2 8 0>,   /* A0 - Reset pin */
+                   <1 0 &gpio2 6 0>,   /* A1 - Busy pin */
+                   <2 0 &gpio1 11 0>,  /* A2 - Event pin */
+                   <3 0 &gpio0 0 0>,   /* A3 - CS pin */
+                   <4 0 &gpio0 4 0>,   /* A4 - Extra GPIO */
+                   <5 0 &gpio0 5 0>,   /* A5 - Extra GPIO */
+                   <6 0 &gpio0 1 0>,   /* D6 - MOSI */
+                   <7 0 &gpio0 2 0>,   /* D7 - MISO */
+                   <8 0 &gpio0 3 0>,   /* D8 - SCK */
+                   <9 0 &gpio2 6 0>,   /* D9 - Busy pin */
+                   <10 0 &gpio0 0 0>,  /* D10 - CS pin */
+                   <11 0 &gpio1 11 0>, /* D11 - Event pin */
+                   <12 0 &gpio0 2 0>,  /* D12 - MISO */
+                   <13 0 &gpio0 0 0>,  /* D13 - CS pin */
+                   <14 0 &gpio0 6 0>,  /* D14 - Extra GPIO */
+                   <15 0 &gpio0 7 0>;  /* D15 - Extra GPIO */
```
It seem not be compatible with Arduino pinout https://docs.arduino.cc/retired/boards/arduino-uno-rev3-with-long-pins/#pinout-diagram

RobertGalatNordic commented on boards/shields/semtech_sx126xmb2xxs/doc/index.rst:
```diff
@@ -0,0 +1,90 @@
+   :board: nucleo_l073rz
+   :shield: semtech_sx126xmb2xxs
+   :goals: build
+
```
I do not know if we need this example in the doc, and even if we do, we could use our own sample to show how to build and not lorawan

RobertGalatNordic commented on boards/shields/semtech_sx126xmb2xxs/semtech_sx126xmb2xxs_common.dtsi:
```diff
@@ -0,0 +1,41 @@
+    };
+};
+
+// FIXME: Is there a way to reassign the node label arduino_spi from spi2 to spi3 ?
```
we can remove this fixme, we can reasign the arduino_spi in the Arduino shield

ktaborowski replied:
Fixed in arduino header @RobertGalatNordic  please have a look on commit: boards: arduino shield board agnostic

RobertGalatNordic commented on doc/semtech_radio_refactor.md:
```diff
@@ -0,0 +1,131 @@
+#### Optional Properties (❌ Not Implemented in Driver Structure)
+| DTS Property | Reason | Description |
+|--------------|--------|-------------|
+| `dio2-gpios` | No field in driver structure | DIO2 as generic IRQ line |
```
dio2-gpios is mutuialy exclusive to rf-switch, so it is disabled by driver

RobertGalatNordic commented on doc/semtech_radio_refactor.md:
```diff
@@ -0,0 +1,131 @@
+| `dio2-as-rf-switch` | No field in driver structure | DIO2 as RF switch control |
+| `dio3-gpios` | No field in driver structure | DIO3 as generic IRQ line |
+| `dio3-as-tcxo-control` | No field in driver structure | DIO3 as TCXO control |
+| `xtal-capacitor-value-xta` | No field in driver structure | XTA trimming capacitor value |
+| `xtal-capacitor-value-xtb` | No field in driver structure | XTB trimming capacitor value |
```
trim cap can be set in `subsys/config/common/src/app_subGHz_config.c: 89` so we can use DTS value even if there is no field in driver config for it.

RobertGalatNordic commented on doc/semtech_radio_refactor.md:
```diff
@@ -0,0 +1,131 @@
+| `dio3-as-tcxo-control` | No field in driver structure | DIO3 as TCXO control |
+| `xtal-capacitor-value-xta` | No field in driver structure | XTA trimming capacitor value |
+| `xtal-capacitor-value-xtb` | No field in driver structure | XTB trimming capacitor value |
+| `tx-power-offset` | No field in driver structure | TX power offset configuration |
```
OK this value is really not used in sidewalk driver, so it is correct to say that it will be ignored.

RobertGalatNordic reviewed (approved):
lgtm, maybe some minor comments on the documentation, but generally it seems as a change in good direction.

I see that some tests failed. We need to take a look at them, and I think it should be simple alignment of the configuration in test builds, but this has to be handled before merge.

RobertGalatNordic commented on samples/sid_end_device/boards/nrf52840dk_nrf52840.overlay:
```diff
@@ -4,35 +4,7 @@
  * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
  */
 
-&spi3 {
-	status = "disabled";
-};
-
-&spi2 {
-	status = "disabled";
-};
-
```
if we bring this back, and revert deletion of bus_nrf_spi.c we could still have the nrfx implementation on nrf52840 dk.

We could add some kconfig warning that if someone build with the semetech shield on this platform this is not supported.

but I think that by simply reverting deletion of this file and deletion of nrfx_spi.c we could retain the old working behaviour on the nrf52 platform

ktaborowski committed:
doc: semtech shield

Add documentation

Signed-off-by: Krzysztof Taborowski <krzysztof.taborowski@nordicsemi.no>

ktaborowski committed:
tests: use semtech from shields

Fix spi test, to get the right spi instance.
Other minor fixes for ci.

Signed-off-by: Krzysztof Taborowski <krzysztof.taborowski@nordicsemi.no>

ktaborowski committed:
config: add semtech as a shield

Refactor semtech radio support.
Add shields (dts) like in Lora Basic Modem
project for Zephyr.

Signed-off-by: Krzysztof Taborowski <krzysztof.taborowski@nordicsemi.no>

ktaborowski committed:
samples: use semtech shield is samples

Use refactored semtech shields approach.

Signed-off-by: Krzysztof Taborowski <krzysztof.taborowski@nordicsemi.no>

ktaborowski committed:
samples: add nrf spi snippet

Add snippet for nrfx spi driver

Signed-off-by: Krzysztof Taborowski <krzysztof.taborowski@nordicsemi.no>

RobertGalatNordic committed:
build: fix issue with multiple peripherals on one block

previous shield implementation assigned multiple
peripherals into single hardware block
which is not valid configuration

removed the unused interfaces from arduino adapter.
if there is need for them in the future they can be added as needed.

Signed-off-by: Robert Gałat <robert.galat@nordicsemi.no>

ktaborowski commented on boards/shields/simple_arduino_adapter/boards/nrf54l15dk_common.overlay:
```diff
@@ -40,27 +43,30 @@
-    pinctrl-1 = <&uart21_sleep_alt>;
-    pinctrl-names = "default", "sleep";
-};
+// arduino_i2c: &i2c22 {
+//     status = "okay";
+//     pinctrl-0 = <&i2c22_default_alt>;
+//     pinctrl-1 = <&i2c22_sleep_alt>;
+//     pinctrl-names = "default", "sleep";
+// };
```
Commented code, please uncomment or remove
